runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -4780,6 +4780,18 @@ TR_J9InlinerPolicy::supressInliningRecognizedInitialCallee(TR_CallSite* callsite
          return true;
       case TR::java_lang_Class_cast:
          return true; // Call will be transformed into checkcast
+      case TR::java_lang_Object_hashCode:
+         if (callNode &&
```

Comment:
Is it assumed that this call node will have a direct call opcode ? Otherwise the first child will be a load of the vft, not the actual first argument.

It is probably better to check `getNumChildren() > 0` in order to determine that one can access the first child. It is possible in theory that some transformation sets the number of children to N and does not null out the child pointers for any value above N.